frontend/src/components/unsubscribe/Unsubscribe.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from 'react'
+import { useLocation, useParams, Redirect } from 'react-router-dom'
+import querystring from 'querystring'
+
+import { ErrorBlock, PrimaryButton } from 'components/common'
+
+import styles from './Unsubscribe.module.scss'
+import appLogo from 'assets/img/brand/app-logo.svg'
+import landingHero from 'assets/img/landing/landing-hero.png'
+
+import {
+  unsubscribeRecipient,
+  isUserUnsubscribed,
+} from 'services/unsubscribe.service'
+
+const Unsubscribe = () => {
+  const { version } = useParams()
+  const location = useLocation()
+  const [errorMsg, setErrorMsg] = useState('')
+  const [isValid, setValid] = useState(true)
+  const [isUnsubscribed, setUnsubscribed] = useState(false)
+  const [campaignId, setCampaignId] = useState('')
+  const [recipient, setRecipient] = useState('')
+  const [hash, setHash] = useState('')
+
+  async function onConfirmation() {
+    try {
+      await unsubscribeRecipient({
+        campaignId: +campaignId,
+        recipient,
+        hash,
+        version,
+      })
+      setUnsubscribed(true)
+    } catch (err) {
+      setErrorMsg(err.message)
+    }
+  }
+
+  async function isUrlValid({
+    campaignId,
+    recipient,
+    hash,
+    version,
+  }: {
+    campaignId: number
+    recipient: string
+    hash: string
+    version: string
+  }) {
+    try {
+      const isUnsub = await isUserUnsubscribed({
+        campaignId: +campaignId,
+        recipient,
+        hash,
+        version,
+      })
+
+      setUnsubscribed(isUnsub)
+      setValid(true)
+      return
+    } catch (e) {
+      setValid(false)
+    }
+  }
+
+  useEffect(() => {
+    // Remove the question mark infront
+    // ?c=32&r=shaowei%40open.gov.sg&h=82f3 -> c=32&r=shaowei%40open.gov.sg&h=82f3
+    const query = location.search.substr(1)
+    const { c, r, h } = querystring.parse(query)
```

Comment:
Having to remove the `?` and checking if they are of type `string` seems weird to me.

Reply:
```javascript
a = 'https://postman.gov.sg/v1/unsubscribe?c=aaa&r=bbb&h=ddd&h=eee'
g = querystring.parse(url.parse(a).query)
isValid = Object.values(g).every(x=>typeof x === 'string')
```

Reply:
Thanks that's alot more elegant than my implementation